nimbleparse/src/main.rs

```diff
@@ -176,7 +176,35 @@ fn main() {
 
     let yacc_y_path = PathBuf::from(&matches.free[1]);
     let yacc_src = read_file(&yacc_y_path);
-    let ast_validation = ASTWithValidityInfo::new(&mut header, &yacc_src);
+    let yk_val = header.get("yacckind");
+    if yk_val.is_none() {
+        let parsed_header = GrmtoolsSectionParser::new(&yacc_src, true).parse();
+        match parsed_header {
+            Ok((parsed_header, _)) => {
+                header
+                    .merge_from(parsed_header)
+                    .expect("Specified merge behavior cannot fail");
+            }
+            Err(errs) => {
+                eprintln!(
+                    "Errors parsing `%grmtools` section:\n {}\n",
+                    errs.iter()
+                        .map(|e| e.to_string())
+                        .collect::<Vec<_>>()
+                        .join("\n")
+                );
+                std::process::exit(1);
+            }
+        }
+    }
+    let yk_val = header.get("yacckind");
+    if yk_val.is_none() {
+        eprintln!("Cannot determine yacckind, add it to the `%grmtools` section in the yacc file, or pass the `-y` argument");
```

Comment:
I might phrase this one as "yacckind not specified in the `%grmtools` section of the grammar or via the '-y' parameter", just to try and fit a typical terminal width a bit better.